src/Common/src/Interop/Windows/secur32/ApplicationProtocols.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Net
+{
+    // _SEC_APPLICATION_PROTOCOLS
+    [StructLayout(LayoutKind.Sequential,Pack = 1)]
+    internal struct ApplicationProtocols
+    {
+        private static readonly int ProtocolListOffset = Marshal.SizeOf<ApplicationProtocols>();
+        private static readonly int ProtocolListConstSize = ProtocolListOffset - (int)Marshal.OffsetOf<ApplicationProtocols>("ProtocolExtenstionType");
+        public uint ProtocolListsSize;
+        public ApplicationProtocolNegotiationExt ProtocolExtenstionType;
+        public short ProtocolListSize;
+
+        public static unsafe byte[] ToByteArray(string[] applicationProtocols)
```

Comment:
There are argument exceptions being thrown if applicationProtocols contains nulls... what if it is null?